src/foam/nanos/pm/cronjobs.jrl

```diff
@@ -0,0 +1 @@
+p({"class":"foam.nanos.cron.Cron","id":"PMInfoRollup","enabled":true,code":"import foam.dao.DAO;\nimport foam.nanos.logger.Logger;\nimport foam.nanos.pm.PM;\nimport foam.nanos.pm.PMInfo;\nimport foam.nanos.pm.DAOPMLoggerCopySink;\nimport foam.nanos.script.Script;\n\nscriptName = \"PMInfoRollup\";\npm = new PM(Script.class, \"scriptName\");\nlogger = x.get(\"logger\");\nlogger.info(new String[] {scriptName, \"start\"});\n\n// current 1 minute dao\ndao = x.get(\"pmInfoDAO\").getDelegate();\n\n// replace with new empty dao\nx.get(\"pmInfoDAO\").setDelegate(new foam.dao.MDAO(foam.nanos.pm.PMInfo.getOwnClassInfo()));\n\none = x.get(\"pmInfo1minDAO\");\nfive = x.get(\"pmInfo5minDAO\");\nhour = x.get(\"pmInfo1hourDAO\");\nday = x.get(\"pmInfo1dayDAO\");\n\n//logger.info(new String[] {scriptName, \"daos\", one.toString(), five.toString(), hour.toString(), day.toString()});\n//logger.info(new String[] {scriptName, \"current replaced\"});\n\n// move previous 1 minute to named 1 minute dao\none.setDelegate(dao);\n\n// cleanup\nnow = java.time.LocalDateTime.now();\n\n// if at 5 minute interval of hour clear 5 minute dao\nif ( now.getMinute() % 5 == 0 ) {\n  five.removeAll();\n  logger.info(new String[] {scriptName,\"clean 5 minute\"});\n}  \n\n// if at top of the hour, clear hour dao, if start of day.\nif ( now.getMinute() == 0 ) {\n  hour.removeAll();\n  logger.info(new String[] {scriptName,\"clean 1 hour\"});\n  if ( now.getHour() == 0 ) {\n    day.removeAll();\n    logger.info(new String[] {scriptName,\"clean 1 day\"});\n  }\n}\n\n// put 1 minute entries into other daos.\nDAO[] daos = new DAO[] { five, hour, day };\ndao.select(new DAOPMLoggerCopySink(x, daos));\nlogger.info(new String[] {scriptName,\"end\"});\npm.log(x);\n","output":""})
```

Comment:
There's a syntax error in here. Looks like a problem with double quote characters not being escaped properly.

```
2019.04.29.09.09.31.005,SEVERE,[JDAO],cronjobs.0,Parse error,Invalid character '"' found at 71
Valid characters include: 'N','k','J','g','F','c','B','_','[','x','W','6','t','S','2','p','O','l','K','h','G','d','C','`','\','y','X','7','u','T','3','q','P','m','L','i','H','e','D','a',']','z','Y','8','v','U','4','r','Q','0','n','M','j','I','f','E','$','b','A','^','Z','9','w','V','5','s','R','1','o',line:,{"class":"foam.nanos.cron.Cron","id":"PMInfoRollup","enabled":true,code":"import foam.dao.DAO;\nimport foam.nanos.logger.Logger;\nimport foam.nanos.pm.PM;\nimport foam.nanos.pm.PMInfo;\nimport foam.nanos.pm.DAOPMLoggerCopySink;\nimport foam.nanos.script.Script;\n\nscriptName = \"PMInfoRollup\";\npm = new PM(Script.class, \"scriptName\");\nlogger = x.get(\"logger\");\nlogger.info(new String[] {scriptName, \"start\"});\n\n// current 1 minute dao\ndao = x.get(\"pmInfoDAO\").getDelegate();\n\n// replace with new empty dao\nx.get(\"pmInfoDAO\").setDelegate(new foam.dao.MDAO(foam.nanos.pm.PMInfo.getOwnClassInfo()));\n\none = x.get(\"pmInfo1minDAO\");\nfive = x.get(\"pmInfo5minDAO\");\nhour = x.get(\"pmInfo1hourDAO\");\nday = x.get(\"pmInfo1dayDAO\");\n\n//logger.info(new String[] {scriptName, \"daos\", one.toString(), five.toString(), hour.toString(), day.toString()});\n//logger.info(new String[] {scriptName, \"current replaced\"});\n\n// move previous 1 minute to named 1 minute dao\none.setDelegate(dao);\n\n// cleanup\nnow = java.time.LocalDateTime.now();\n\n// if at 5 minute interval of hour clear 5 minute dao\nif ( now.getMinute() % 5 == 0 ) {\n  five.removeAll();\n  logger.info(new String[] {scriptName,\"clean 5 minute\"});\n}  \n\n// if at top of the hour, clear hour dao, if start of day.\nif ( now.getMinute() == 0 ) {\n  hour.removeAll();\n  logger.info(new String[] {scriptName,\"clean 1 hour\"});\n  if ( now.getHour() == 0 ) {\n    day.removeAll();\n    logger.info(new String[] {scriptName,\"clean 1 day\"});\n  }\n}\n\n// put 1 minute entries into other daos.\nDAO[] daos = new DAO[] { five, hour, day };\ndao.select(new DAOPMLoggerCopySink(x, daos));\nlogger.info(new String[] {scriptName,\"end\"});\npm.log(x);\n","output":""}
```